framework/app.mk

```diff
@@ -429,7 +429,18 @@ ifneq ($(app_test_LIB),)
 	lib_install_targets += $(dir $(app_test_LIB))install_lib_$(notdir $(app_test_LIB))
 endif
 
-install_libs: $(lib_install_targets)
+install_libs:: $(lib_install_targets)
+
+ifneq ($(wildcard $(APPLICATION_DIR)/data/.),)
+install_data_$(APPLICATION_NAME)_src := $(APPLICATION_DIR)/data
+install_data_$(APPLICATION_NAME)_dst := $(share_install_dir)
+install_data:: install_data_$(APPLICATION_NAME)
+endif
+
+install_data_%:
```

Comment:
No issues here, but we probably should have coordinated a bit. The target below is designed to handle an arbitrary number of directories transformed to an arbitrary number of installed locations. I designed it that way so that applications like Griffin could function. Yes, they have multiple data directories and may not want to collapse them all into one. I'll chat with @YaqiWang, but this could be a bit of a showstopper from implementing it exactly they way you have it here.

Reply:
As discussed on Slack, there can still be a hierarchy below `data`. The folders wouldn't have to get flattened.